workflow/controller/operator.go

```diff
@@ -819,8 +837,8 @@ func (woc *wfOperationCtx) podReconciliation() error {
 	// It is now impossible to infer pod status. The only thing we can do at this point is to mark
 	// the node with Error.
 	for nodeID, node := range woc.wf.Status.Nodes {
-		if node.Type != wfv1.NodeTypePod || node.Fulfilled() || node.StartedAt.IsZero() {
-			// node is not a pod, it is already complete, or it can be re-run.
+		if node.Type != wfv1.NodeTypePod || node.Fulfilled() || node.StartedAt.IsZero() || ExceededQuota(&node) || FailedQuota(&node) {
```

Comment:
This would dead-lock the controller, making the solution in https://github.com/argoproj/argo/pull/2385 moot.